src/services/expenses.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const { Expense } = require('../models/Expense.js');
+
+async function getAll({ userId, categories, from, to }) {
+  const expenses = await Expense.findAll({
+    order: [ 'createdAt' ],
+  });
+
+  const filteredExpenses = expenses.filter(expense => {
+    const byUserId = userId
+      ? expense.userId === +userId
+      : true;
+
+    const byCategories = categories
+      ? categories.includes(expense.category)
+      : true;
+
+    const byFrom = from
+      ? expense.spentAt > from
```

Comment:
expense.spentAt > from already return true or false

Reply:
I use 'true' in the filter in case some parameter is not passed. Because in this case all the data are suitable.